scripts/kubestellar-snapshot.sh

```diff
@@ -0,0 +1,554 @@
+    (-X)
+        set -x;;
+    (-h|--help)
+        display_help
+        exit 0;;
+    (-*)
+        echo "$0: unknown flag" >&2
+        exit 1;;
+    (*)
+        echo "$0: unknown positional argument" >&2
+        exit 1;;
+    esac
+    shift
+done
+
+
+###############################################################################
+# Define alias definition
+###############################################################################
+# Define the echov function based on verbosity
+if [ "$arg_verbose" == "true" ]; then
+    echov() { echo "$@" ; }
+else
+    echov() { :; }
+fi
+
+# Define the special kubectl
+k() {
+    KUBECONFIG="$kubeconfig" kubectl $*
+}
+
+
+###############################################################################
+# Script info
+###############################################################################
+echov -e "${COLOR_INFO}$($0 --version)${COLOR_NONE}\n"
+echov -e "Script run on ${COLOR_INFO}$timestamp${COLOR_NONE}"
+
+
+###############################################################################
+# Check dependencies
+###############################################################################
+echov "Checking script dependencies:"
+
+is_installed 'kubectl' \
+    'kubectl' \
+    'kubectl version --client | head -1 | cut -d" " -f3'
+
+is_installed 'helm' \
+    'helm' \
+    'helm version | cut -d"\"" -f2'
+
+is_installed 'jq' \
+    'jq' \
+    'jq --version'
+
+
+###############################################################################
+# Ensure output folder
+###############################################################################
+if [[ "$arg_logs" == "true" || "$arg_yaml" == "true" ]] ; then
+    yes "yes" | rm -vRI "/tmp/kubestellar-snapshot" > /dev/null 2>&1 || true
+    mkdir -p "$output_folder"
+fi
+
+
+###############################################################################
+# Determine the list of kubeconfigs to search
+# based on https://kubernetes.io/docs/reference/k/generated/k_config/
+###############################################################################
+if [[ "$arg_kubeconfig" != "" ]] ; then
+    kubeconfig="$arg_kubeconfig"
+elif [[ "$KUBECONFIG" != "" ]] ; then
+    kubeconfig="$KUBECONFIG"
+else
+    kubeconfig="$HOME/.kube/config"
+fi
+echov -e "Using kubeconfig(s): ${COLOR_INFO}$kubeconfig${COLOR_NONE}"
+
+
+###############################################################################
+# Determine the list of contexts to search
+###############################################################################
+if [[ "$arg_context" == "" ]] ; then
+    contexts=($(k config get-contexts --no-headers -o name))
+else
+    contexts=("$arg_context")
+fi
+
+if [[ -z "${contexts[@]}" ]] ; then
+    echoerr "no context(s) found in the kubeconfig file $kubeconfig!"
+    exit 1
+else
+    echov "Using contexts(s): "
+    for context in "${contexts[@]}" ; do # for all contexts
+        if k --context $context get pods -A  > /dev/null 2>&1 ; then
```

Comment:
All of our current spaces have a resource named "pods" but that I something that we hope to change. The "namespace" resource, OTOH, I would expect in every space.

Reply:
Another thought: every Space will have the "secrets" resource. Ones that fail to list secrets will also fail (and taking another long time to fail, if the "get secrets" does) to do the `helm list` command later. So perhaps the better choice here is to try `get secrets` --- and also compile a list of contexts where that succeeded, and do the `helm list` only in those contexts.

Reply:
I changed to use `secrets`